plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -268,3 +282,79 @@ func (cache *schedulerCache) expirePod(key string, ps *podState) error {
 	delete(cache.podStates, key)
 	return nil
 }
+
+func (cache *schedulerCache) AddGroupingObject(obj interface{}) error {
+	switch t := obj.(type) {
+	case *api.Service:
+		objRef := &api.ObjectReference{Kind: "Service", Namespace: t.Namespace, Name: t.Name}
+		selector := labels.Set(t.Spec.Selector).AsSelector()
+		return cache.addGroupingObject(objRef, selector)
+	default:
+		return fmt.Errorf("unsupported object type: %v", t)
+	}
+	return nil
+}
+
+func (cache *schedulerCache) addGroupingObject(ref *api.ObjectReference, selector labels.Selector) error {
+	// TODO: Implement:
+	// 1. Lock the cache
+	// 2. Choose all pods satisfying the selector
+	// 3. For each of them:
+	//    (a) add it to mapping grouping object -> its pods
+	//    (b) if the pod wasn't already there (it shouldn't, but races?) -> add +1 for NodeInfo for this objRef.
+	return nil
+}
+
+func (cache *schedulerCache) UpdateGroupingObject(oldObj, newObj interface{}) error {
+	switch t := newObj.(type) {
+	case *api.Service:
+		switch t2 := oldObj.(type) {
+		case *api.Service:
+			if reflect.DeepEqual(t.Spec.Selector, t2.Spec.Selector) {
```

Comment:
I'm not aware of any - we can create a map comparator if needed. I'm just trying to prototype it to see the impact of it.